src/util.c

```diff
@@ -690,24 +689,13 @@ static int hyper_open_virtio_port(char *channel, int mode)
 			continue;
 		}
 
-		fprintf(stdout, "open hyper channel %s\n", path);
-		fd = open(path, O_RDWR | O_CLOEXEC | mode);
-		if (fd < 0)
-			perror("fail to open channel device");
-
-		break;
+		fprintf(stdout, "find hyper channel %s\n", path);
+		free(list);
```

Comment:
The dirent array is not freed <- the original code has the same issue though.

Reply:
@bergwolf good catch, updated!